### PR TITLE
support regression test with manifest file

### PR DESCRIPTION
**Pull request by PengTian0**
**Background**:

The script test.sh doesn't support clone specified version of packages.
However, the regression test for release should test the specified version which is going to be released.
This PR enables the script to meet the demand.

**Test**:
http://rackhdci.lss.emc.com/job/BuildRelease/job/Test/job/RegressionFunctional/
http://rackhdci.lss.emc.com/job/BuildRelease/job/Test/job/RegressionOSInstall/

### CODE REVIEW

**Comment by PengTian0** [comment]
@panpan0000 @changev @anhou @yyscamper

File: test.sh [hunk 1]
```diff
@@ -58,22 +58,86 @@ dlTftpFiles() {
...
-     cd ${WORKSPACE}/build-deps
-     if [ "${i}" != "${REPO_NAME}" ]; then
-         git clone https://github.com/RackHD/${i}.git
-         cd ${i}
-         git checkout ${GIT_REFSPEC}
-         npm install --production
-     fi
-  done
+preparePackages() {
+  if [ -z ${MANIFEST_FILE_URL} ];
```
Comment (panpan0000) [inline test.sh:53]: comments , please.

File: test.sh [hunk 2]
```diff
@@ -58,22 +58,86 @@ dlTftpFiles() {
...
+    sed -i "/mkdir/d" make-deb.sh
+    sudo bash make-deb.sh
+    popd
+    for package in ${API_PACKAGE_LIST}; do
+      sudo pip uninstall -y ${package//./-}
+      pushd ${WORKSPACE}/on-http/$package
+        fail=true
+        while $fail; do
+          sudo python setup.py install
+          if [ $? -eq 0 ];then
```
Comment (panpan0000) [inline test.sh]: nice

File: test.sh [hunk 3]
```diff
@@ -58,22 +58,86 @@ dlTftpFiles() {
...
+  preparePackages
   dlTftpFiles
   dlHttpFiles
+  apiPackageModify
 }
 
+apiPackageModify() {
+  if [[ "${REPO_NAME}" == *"on-http"* ]] && [[ "${REPO_NAME}" == *"rackhd"* ]]; then
+    pushd ${WORKSPACE}/on-http/extra
+    sed -i "s/.*git symbolic-ref.*/ continue/g" make-deb.sh
```
Comment (panpan0000) [inline test.sh]: seems you are replace the $BRANCH in on-http/extra/make-deb.sh, but can you please add comment in code ?

File: test.sh [hunk 4]
```diff
@@ -58,22 +58,86 @@ dlTftpFiles() {
...
+    sed -i "/build-package.bash/d" make-deb.sh
+    sed -i "/mkdir/d" make-deb.sh
+    sudo bash make-deb.sh
+    popd
+    for package in ${API_PACKAGE_LIST}; do
```
Comment (panpan0000) [inline test.sh]: where does the $API+PACKAGE_LIST come from ?

File: test.sh [hunk 5]
```diff
@@ -58,22 +58,86 @@ dlTftpFiles() {
...
+
+prepareDeps(){
+  preparePackages
   dlTftpFiles
   dlHttpFiles
+  apiPackageModify
 }
 
+apiPackageModify() {
+  if [[ "${REPO_NAME}" == *"on-http"* ]] && [[ "${REPO_NAME}" == *"rackhd"* ]]; then
```
Comment (panpan0000) [inline test.sh]: why not only ```if [[ "${REPO_NAME}" == *"on-http"* ]] ; then```

File: test.sh [hunk 6]
```diff
@@ -23,8 +15,12 @@ if [ ! -z "${2}" ]; then
 fi
 
 dlHttpFiles() {
-  dir=${WORKSPACE}/build/static/http/common
-  if [ "${REPO_NAME}" != "on-http" ]; then
+  if [ -n "${MULTI}" ]; then
```
Comment (panpan0000) [inline test.sh]: I didn't find $MULTI  in your test job ?

how about add some comment to help review ?

File: test.sh [hunk 7]
```diff
@@ -41,8 +37,12 @@ dlHttpFiles() {
...
 
 dlTftpFiles() {
-  dir=${WORKSPACE}/build/static/tftp
-  if [ "${REPO_NAME}" != "on-tftp" ]; then
+  if [ -n "${MULTI}" ]; then
+    dir=${WORKSPACE}/on-tftp/static/tftp
+  else
+    dir=${WORKSPACE}/build/static/tftp
+  fi
+  if [[ "${REPO_NAME}" != *"on-tftp"* ]]; then
```
Comment (panpan0000) [inline test.sh]: why do we need the star (*) ? to support what kind of "$REPO_NAME" instance ? can you give an example ?

**Review by panpan0000 — APPROVED** [review]
I believe the script is workable based on your experimental jenkins job .
but I still insist more **comments** and explanation, especially the tricky parts like the "sed" operation...etc

:+1:  to move on the release schedule .

**Comment by changev** [comment]
test this please

**Comment by PengTian0** [comment]
@panpan0000 @changev

**Comment by panpan0000** [comment]
:+1:  now. good job

**Comment by changev** [comment]
LGTM  :+1:

**Review by yyscamper — APPROVED** [review]

**Pull request merged**